### PR TITLE
Water Gun

### PR DESCRIPTION
**Pull request by SapphicOverload**
# Document the changes in your pull request

Adds a water gun to the game, the donksoft hyper-soaker! You fill it with water, then shoot people with it to splash them. A harmless prank unless you're allergic to water or something, though it can be filled with other reagents too. Available from hacked donksoft vending machines, or from a syndicate uplink for 2tc.

Standard version holds 250 units and splashes 10 units at a time, uplink version holds 600 units and splashes 30 at a time.

Also adjusts preternis water damage to be more affected by smaller amounts of water so that this actually does something to them, and updates it on each splash. A few other things have been refactored too but don't affect gameplay.

# Why is this good for the game?

I thought it would be funny, also has some neat interactions with chemistry that will make things more interesting, and having a counter to preternis like there is to IPCs is probably a good thing.

# Testing

image

# Spriting

Sprite made by @BlueishTsunami
image
image

# Wiki Documentation

Syndicate version should be added to the list of syndicate items

# Changelog

:cl: sapphicoverload, bluishtsunami
rscadd: adds a water gun to donksoft vending machines (and uplink)
tweak: preternis water damage tweaked to be more affected by smaller volumes
tweak: preternis water damage updates immediately when splashed instead of waiting up to 2 seconds
/:cl:

### CODE REVIEW

**Review by cuackles — CHANGES_REQUESTED** [review]
the sprite is missing the inhand icon and the item icon needs more defined outlines

**Comment by Yarinoi** [comment]
i cant wait to supersoak people with fluorosulphuric acid

**Comment by ju45he** [comment]
Use this and lexorin and an ided nerf will get speed merged faster than you can say "clown port aft maints linging AI".

**Comment by iamthesilly** [comment]
> Use this and lexorin and an ided nerf will get speed merged faster than you can say "clown port aft maints linging AI".

Lexorin has no effect in this gun since it uses touch application and most chems don't do anything on touch application

**Comment by ynot01** [comment]
do what cuackles said

File: yogstation/code/modules/mob/living/carbon/human/species_types/preternis/preternis.dm [hunk 1]
```diff
@@ -238,6 +238,7 @@
 		H.adjust_wet_stacks(-1)
 		soggy = TRUE
 		H.throw_alert("preternis_wet", /atom/movable/screen/alert/preternis_wet)
+		H.adjust_fire_stacks(1)
```
Comment (Moltijoe) [inline yogstation/code/modules/mob/living/carbon/human/species_types/preternis/preternis.dm]: shouldn't need this line anymore

**Review by Moltijoe — CHANGES_REQUESTED** [review]

**Review by Moltijoe — APPROVED** [review]

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 2]
```diff
@@ -121,7 +121,7 @@
 /datum/reagent/water
 	name = "Water"
 	description = "An ubiquitous chemical substance that is composed of hydrogen and oxygen."
-	color = "#AAAAAA77" // rgb: 170, 170, 170, 77 (alpha)
+	color = "#609bdf77" // rgb: 170, 170, 170, 77 (alpha)
```
Comment (ynot01) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: Change comment to 96, 155, 223, 77

File: code/modules/projectiles/projectile/special/reagent.dm [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	if(!reagents.total_volume) // if it didn't already have reagents in it, fill it with the default reagents
+		for(var/type in reagents_list)
+			reagents.add_reagent(type, reagents_list[type])
+	add_atom_colour(mix_color_from_reagents(reagents.reagent_list), FIXED_COLOUR_PRIORITY)
+
+/obj/projectile/reagent/fire(angle, atom/direct_target)
+	update_reagents()
+	return ..()
+
+/obj/projectile/reagent/on_hit(atom/target, blocked = FALSE)
```
Comment (ynot01) [inline code/modules/projectiles/projectile/special/reagent.dm:28]: Add a combat log that lists the reagents injected

File: code/game/objects/items/mop.dm [hunk 4]
```diff
@@ -21,7 +21,7 @@
 
 /obj/item/mop/Initialize(mapload)
 	. = ..()
-	create_reagents(mopcap)
+	create_reagents(mopcap, REFILLABLE)
```
Comment (ynot01) [inline code/game/objects/items/mop.dm:24]: Reason for this?

**Review by ynot01 — CHANGES_REQUESTED** [review]

File: code/game/objects/items/mop.dm (hunk 4 above)
Comment (iamthesilly) [reply code/game/objects/items/mop.dm:24]: In order to allow sinks to refill the gun, I changed sinks to refill items based on whether or not they have a refillable reagent holder, so I also made mops refillable now that they don't need a special check anymore.

**Review by ynot01 — APPROVED** [review]

**Comment by ToasterBiome** [comment]
Reopen when work continues

**Review by cuackles — APPROVED** [review]
yipeeeeeeeeee

**Pull request merged**